_posts/2023-11-20-unlocking-the-secrets-to-ingestion.md

```diff
@@ -0,0 +1,191 @@
+
+```json
+POST index-name/_settings
+{
+	"index": {
+		"refresh_interval": "30s"
+	}
+}
+```
+
+As we mentioned in the previous section we increase this delay so that we are sending segments less frequently. They are much larger than the original documents so sending them more quickly could end up saturating the connection of a node.
+
+# Compression Settings
+
+When you are storing large amounts of data it can be handy to use compression to minimize on disk size. We have a few different compression algorithms that you can use but the one we will talk about is Zstandard compression (ZSTD). This compression works well because it is tunable, meaning it lets you prioritize either speed or level of compression. Below is the setting we can use to enable ZSTD compression. Note, the compression level is a default of 3 with 6 being the most compressed and 1 being the least.
+
+```json
+POST index-name/_settings
+{
+	"index": {
+		"codec": "zstd_no_dict",
+		"codec.compression_level": 3
+	}
+}
+```
+
+# End to End
+
+Here are the overall settings that I used (excluding the host settings as these aren’t applied through the cluster API’s). On the client side we are using the asynchronous python client to ensure we can saturate our network connection.
+
+For cluster settings, we balance the primary shards and enable segment replication back pressure. The back pressure setting prevents us from overwhelming the nodes if they fall behind the primary replica.
+
+```json
+{
+  "persistent": {
+    "cluster.routing.allocation.balance.prefer_primary": True,
+    "segrep.pressure.enabled": True
+  }
+}
+```
+
+Then for our index, we increase the number of shards (and have only one replica for them). We turn on ZSTD compression to decrease the size on disk. Finally, we turn on segment replication and increase the refresh interval.
+
+```json
+POST index-name/_settings
+{
+  "index": {
+    "number_of_shards": 3,
+    "number_of_replicas": 1,
+    "codec": "zstd_no_dict",
+    "replication.type": "SEGMENT",
+    "refresh_interval": "30s"
+	}
+}
+```
+
+With all of these settings we were able to see a 65% improvement in ingestion speed and a 19% reduction of the size of our data on disk. These are some pretty serious gains for just doing some straightforward settings updates. While your experience may be different you now can tune your ingestion to fit the needs of your dataset! For more information, check out [Tuning your cluster for indexing speed](https://opensearch.org/docs/latest/tuning-your-cluster/performance/) for a deeper dive on some of these strategies.
+
+If enjoyed this post and wanted to see it with some more detail you can check out my presentation from OpenSearchCon 2023 that this blog was drawn from:
+
+{% include youtube-player.html id=page.ingestingToOpenSearch %}
```

Comment:
Edit to image DocumentReplication.svg - Change "Out of the Box" to "Out-of-the-box".

Reply:
Resolved, thank you!